### PR TITLE
Oracle RAC Image

### PR DESCRIPTION
**Pull request by psaini79**
Created the fork from oracle/docker-images and added following psaini79/docker-images:
OracleConnectionManager
OracleRealApplicationCluster
OracleRACStorageServer

### CODE REVIEW

File: OracleDatabase/OracleConnectionManager/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        For complete Oracle Connection Manager setup, please go though following steps and execute them as per your enviornment:
+
+2) You need to make sure that you have enough memory and cpu resources available for container.
+
+3) Create Oracle Connection Manager Image
+
+                IMPORTANT: You will have to provide the installation binaries of Oracle ADMIN Client Oracle Database 12c Release 2 Client (12.2.0.1.0) for Linux x86-64 and put them into the (DOCKER_CMAN_IMAGE)/dockerfiles/(VERSION) folder. You  only need to provide the binaries for the edition you are going to install. The binaries can be downloaded from the Oracle Technology Network. You also have to make sure to have internet connectivity for yum. Note that you must not uncompress the binaries. The script will handle that for you and fail if you uncompress them manually!
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/README.md]: The path seems wrong here. Also, what script are you referring to? You need to be able to build this image without using `buildDockerImage.sh`

File: OracleDatabase/OracleConnectionManager/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        Before you build the image make sure that you have provided the installation binaries and put them into the right folder. Go into the dockerfiles folder and run the buildDockerImage.sh script as root or with sudo privileges:
+
+        Change your directory to (DOCKER_CMAN_IMAGE)/dockerfile folder and execute following command:
+
+        #./buildDockerImage.sh -v (Software Version)
+
+        #./buildDockerImage.sh -v 12.2.0.1
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/README.md]: How do I build the image without using the script?

File: OracleDatabase/OracleConnectionManager/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        For detailed usage of command, please execute folowing command: [oracle@localhost dockerfiles]$ ./buildDockerImage.sh -h
+
+4) Before creating container, create the bridge. Also, replace IP according to your environment. If you are using same bridge with same network then you can use same IPs mentioned in next step.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/README.md]: How do I know what IP to use? What is the next step?

File: OracleDatabase/OracleConnectionManager/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+        #docker network create --driver=bridge --subnet=172.15.1.0/24 rac_pub1_nw
+
+5) Create Containers. We are creating container in non-priv mode.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/README.md]: You don't need to mention "non-priv" here.

File: OracleDatabase/OracleConnectionManager/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        --network=rac_pub1_nw --ip=172.15.1.15 \
+        -e DOMAIN=example.com -e PUBLIC_IP=172.15.1.15 \
+        -e PUBLIC_HOSTNAME=racnode-cman1 -e SCAN_NAME=racnode-scan \
+        -e SCAN_IP=172.15.1.70 --privileged=false \
+        -p 1521:1521 --name racnode-cman oracle/client-cman:12.2.0.1
+
+        In the above container, you can see that we are passing env variables using "-e". You need to change PUBLIC_IP, PUBLIC_HOSTNAME, SCAN_NAME, SCAN_IP according to your environment. Also, container will be binding to port 1521 on your docker host.
+        To check the Cman container/services creation logs , please tail docker logs. It will take 2 minutes to create the Cman container service.
+
+        #docker logs racnode-cman
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/README.md]: There needs to be a space between the # and `docker`

File: OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+#
+# Copyright (c) 1982-2018 Oracle and/or its affiliates. All rights reserved.
+#
+# ORACLE DOCKERFILES PROJECT
+# --------------------------
+# This is the Dockerfile for Oracle Connection Manager
+# 
+# REQUIRED FILES TO BUILD THIS IMAGE
+# ----------------------------------
+# (1) linuxx64_12201_client
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/Dockerfile]: Please specify the exact files needed to be downloaded.

File: OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+#     Download Oracle Connection Manager as part of the Oracle Database Client
+#     from http://www.oracle.com/technetwork/database/enterprise-edition/downloads/index.html
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/Dockerfile]: You probably mean http://www.oracle.com/technetwork/database/enterprise-edition/downloads/oracle12c-linux-12201-3608234.html

File: OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+     DB_LD_LIBRARY_PATH=$DB_HOME/lib:/usr/lib:/lib
+
+# Copy binaries
+# -------------
+# Grid Binaries
+COPY $INSTALL_FILE_1 $SETUP_LINUX_FILE $CHECK_SPACE_FILE $CLIENT_RSP $INSTALL_DB_BINARIES_FILE $INSTALL_DIR/install/
+
+COPY $RUN_FILE $CMANORA $CONFIG_CMAN_FILE $FUNCTIONS $INSTALL_DIR/startup/
+
+RUN chmod 777 $INSTALL_DIR/install/*.sh && \
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/Dockerfile]: Do not set `777` on the script files. You should be setting `755` at most.

File: OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+# Install DB software binaries
+USER oracle
+RUN $INSTALL_DIR/install/$INSTALL_DB_BINARIES_FILE EE 
+
+USER root
+RUN $INVENTORY/orainstRoot.sh && \
+    $DB_HOME/root.sh && \
+    rm -rf $INSTALL_DIR/install && \
+    rm -f /etc/security/limits.d/oracle-database-server-12cR2-preinstall.conf && \
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/Dockerfile]: Why are you removing this file?

File: OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/checkSpace.sh [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Since: January, 2018
+# Author: paramdeep.saini@oracle.com
+# Description: Checks the available space of the system.
+# 
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+# 
+
+REQUIRED_SPACE_GB=6
+AVAILABLE_SPACE_GB=`df -B 1G / | tail -n 1 | awk '{print $4}'`
```
Comment (Djelibeybi) [inline OracleDatabase/OracleConnectionManager/dockerfile/12.2.0.1/checkSpace.sh]: Please update this to match the updated single instance test, i.e. `df -PB 1G /`

File: OracleDatabase/OracleRACStorageServer/README.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+Sample Docker build files to facilitate installation, configuration, and environment setup for DevOps users.
+
+IMPORTANT: This image can be used to setup ASM on NFS for RAC. You can skip if you have physical block devices or NAS server for Oracle RAC and Grid. You need to make sure that NFS server container must be up and running for RAC functioning. This image is for only testing purpose.
+
+You need to make sure that you have enough space available for container.
+
+	Note: If you are behind the proxy, you need to set http_proxy env variable based on your enviornment before building the image.
+
+- Change your directory to (DOCKER_RAC_IMAGE)/dockerfile folder and execute following command:
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/README.md]: Is this the right path?

File: OracleDatabase/OracleRACStorageServer/README.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+ 	 #./buildDockerImage.sh -v (Software Version)
+
+	 #./buildDockerImage.sh -v 12.2.0.1
+
+	 For detailed usage of command, please execute folowing command:
+	[oracle@localhost dockerfiles]$ ./buildDockerImage.sh -h
+
+- Before creating container, create the bridge for NFS storage container. Also, replace IP according to your environment.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/README.md]: You need to provide more detail about how to replace the IP.

File: OracleDatabase/OracleRACStorageServer/README.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+IMPORTANT: This image can be used to setup ASM on NFS for RAC. You can skip if you have physical block devices or NAS server for Oracle RAC and Grid. You need to make sure that NFS server container must be up and running for RAC functioning. This image is for only testing purpose.
+
+You need to make sure that you have enough space available for container.
+
+	Note: If you are behind the proxy, you need to set http_proxy env variable based on your enviornment before building the image.
+
+- Change your directory to (DOCKER_RAC_IMAGE)/dockerfile folder and execute following command:
+
+ 	 #./buildDockerImage.sh -v (Software Version)
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/README.md]: Missing a space between # and `./buildDockerImage.sh`

File: OracleDatabase/OracleRACStorageServer/README.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+	 #./buildDockerImage.sh -v 12.2.0.1
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/README.md]: Missing a space between # and `./buildDockerImage.sh`

File: OracleDatabase/OracleRACStorageServer/README.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+	 For detailed usage of command, please execute folowing command:
+	[oracle@localhost dockerfiles]$ ./buildDockerImage.sh -h
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/README.md]: Please remove `[oracle@localhost dockerfiles]`

File: OracleDatabase/OracleRACStorageServer/README.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+
+- Before creating container, create the bridge for NFS storage container. Also, replace IP according to your environment.
+
+         #docker network create --driver=bridge --subnet=192.168.17.0/24 rac_priv1_nw
+       
+- SELINUX must be in permissive mode.
+
+- Create Containers. We are creating container in non-priv mode.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/README.md]: You don't need to mention "non-priv" here.

File: OracleDatabase/OracleRACStorageServer/README.md [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+
+        #docker run -d -t --hostname racnode-storage \
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/README.md]: Missing a space between `#` and `docker`

File: OracleDatabase/OracleRACStorageServer/README.md [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+
+         Create NFS volume for /oradata
+
+ ####################################################
+
+IMPORTANT: The NFS volume must be /oradata which you will export to RAC containers for ASM storage. It will take 10 minutes for setting up NFS server.
+
+- Create NFS volume using following command:
+
+         #docker volume create --driver local \
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/README.md]: Missing a space.

File: OracleDatabase/OracleRACStorageServer/dockerfile/12.2.0.1/Dockerfile [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+
+# Copy binaries
+# -------------
+# Copy Linux setup Files
+COPY $SETUP_LINUX_FILE $SUDO_SETUP_FILE $INSTALL_SCRIPTS/
+
+# Setup Scripts
+COPY $RUN_FILE $EXPORTFILE $SCRIPT_DIR/
+
+RUN  chmod 777 $INSTALL_DIR/install/*.sh && \
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/dockerfile/12.2.0.1/Dockerfile]: Do not use `777`. You should be using `755`.

File: OracleDatabase/OracleRACStorageServer/dockerfile/12.2.0.1/checkSpace.sh [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Since: January, 2018
+# Author: paramdeep.saini@oracle.com, sanjay.singh@oracle.com
+# Description: Checks the available space of the system.
+# 
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+# 
+
+REQUIRED_SPACE_GB=5
+AVAILABLE_SPACE_GB=`df -B 1G / | tail -n 1 | awk '{print $4}'`
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/dockerfile/12.2.0.1/checkSpace.sh]: Update to `df -PB  1G /`

File: OracleDatabase/OracleRACStorageServer/dockerfile/12.2.0.1/runOracle.sh [hunk 21]
```diff
@@ -0,0 +1,172 @@
...
+declare -r TRUE=0
+export REQUIRED_SPACE_GB=55
+export ORADATA=/oradata
+##################################################
+
+check_space ()
+{
+local REQUIRED_SPACE_GB=$1
+
+AVAILABLE_SPACE_GB=`df -B 1G $ORADATA | tail -n 1 | awk '{print $4}'`
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRACStorageServer/dockerfile/12.2.0.1/runOracle.sh]: Update to `df -PB 1G $ORADATA`

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 22]
```diff
@@ -0,0 +1,474 @@
+FOLLOWING FREELY AVAILABLE PATCH IS REQUIRED FOR THIS IMAGE TO WORK
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: This should not be at the top of the README. You can reference this later when you list the binary archives that need to be downloaded. There's no need to mention "freely available" here either.

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 23]
```diff
@@ -0,0 +1,474 @@
...
+Oracle RAC Database on Docker
+================================
+Sample Docker build files to facilitate installation, configuration, and environment setup for DevOps users. For more information about Oracle Database please see the Oracle Grid/RAC Database Online Documentation. (http://docs.oracle.com/en/database/)
+
+How to build and run
+======================
+This project offers sample Docker files for:
+
+- Oracle Database 12c Release 2 Grid Infrastructure (12.2.0.1.0) for Linux x86-64
+- Oracle Database 12c Release 2 (12.2.0.1.0) Enterprise Edition for Linux x86-64
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: No mention of RAC here?

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 24]
```diff
@@ -0,0 +1,474 @@
...
+
+Section 9   : Copyright
+
+
+Section 1 : Pre-requsites for RAC on Docker
+================================
+IMPORTANT: You must make the changes specified in this section (customized for your environment) before you proceed to the next section.
+
+- For better performance, it is good to use BTRFS file system for Docker storage on the Docker host.
+- Install and configure Docker Engine on Oracle Linux 7 to run RAC on Docker.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Please refer to "Oracle Container Runtime for Docker" and link to the documentation: https://docs.oracle.com/cd/E52668_01/E87205/html/index.html

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 25]
```diff
@@ -0,0 +1,474 @@
...
+- Each container that you will deploy as part of your cluster must satisfy the minimum hardware requirements of the RAC and GI software. An Oracle RAC database is a shared everything database. All data files, control files, redo log files, and the server parameter file (SPFILE) used by the Oracle RAC database must reside on shared storage that is accessible by all the Oracle RAC database instances. You must provide block devices shared across the hosts.  If you don't have shared block storage, you can, for testing purpose only, use the Oracle rac-storage-server image to deploy a docker container providing NFS-based sharable storage.  Note: If you are planning to use the RAC storage Container for shared storage, you must create the RAC Storage Container before starting the containers. 
+
+- Allocate block devices of at least 50GB for OCR/Voting and databases files. You can refer Oracle 12.2 GRID and database install guides for details on recommended number of devices and total storage size.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Please link to the install guides.

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 26]
```diff
@@ -0,0 +1,474 @@
...
+	- Public IP address for each container
+
+	- Private IP address for each container
+
+	- Virtual IP address for each container
+
+	- Three single client access name (SCAN) addresses for the cluster.
+
+	For details, please refer to following guide:
+	https://docs.oracle.com/database/122/LADBI/LADBI.pdf
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Please link to the HTML documentation, not the PDF.

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 27]
```diff
@@ -0,0 +1,474 @@
...
+
+- You need to plan your private and public network for containers before you start installation. You can create a network bridge on every host so containers running within that host can communicate with each other.  For example, create rac_pub1_nw for the public network (172.15.1.0/24) and rac_priv1_nw (192.168.17.0/24) for a private network. You can use any network subnet for testing. However, in this document we reference the public network on 172.15.1.0/24 and the private network on 192.168.17.0/24.
+
+		#docker network create --driver=bridge --subnet=172.15.1.0/24 rac_pub1_nw
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: You need a space between # and `docker`

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 28]
```diff
@@ -0,0 +1,474 @@
...
+
+		#docker network create --driver=bridge --subnet=192.168.17.0/24 rac_priv1_nw
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: You need a space between # and `docker`

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 29]
```diff
@@ -0,0 +1,474 @@
...
+
+		Note: You must run  RAC on Docker on multi-host using MACVLAN Docker driver (see Docker documentation).
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: If you need to use the `MACVLAN` driver, then please provide example commands on how to configure this.

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 30]
```diff
@@ -0,0 +1,474 @@
...
+   
+- If the docker bridge network is not available outside your host, you can use the Oracle Connection Manager (CMAN) image to access the RAC Database from outside the host. You need to download and setup Oracle Connection Manager image. For the setup, please refer README of Oracle Connection Manager image.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Please link to the README.md for the Connection Manager image.

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 31]
```diff
@@ -0,0 +1,474 @@
...
+
+		#docker network create --driver=bridge --subnet=172.15.1.0/24 rac_pub1_nw
+
+		#docker network create --driver=bridge --subnet=192.168.17.0/24 rac_priv1_nw
+
+		Note: You must run  RAC on Docker on multi-host using MACVLAN Docker driver (see Docker documentation).
+   
+- If the docker bridge network is not available outside your host, you can use the Oracle Connection Manager (CMAN) image to access the RAC Database from outside the host. You need to download and setup Oracle Connection Manager image. For the setup, please refer README of Oracle Connection Manager image. 
+
+- RAC needs to run certain processes in real time mode. To run processes inside a container in real time mode, you need to make changes in your docker configuration files. Please update OPTIONS value in /etc/sysconfig/docker to following:
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Please use back ticks around `/etc/sysconfig/docker`

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 32]
```diff
@@ -0,0 +1,474 @@
...
+
+	OPTIONS='--selinux-enabled --cpu-rt-runtime=950000 --cpu-rt-period=1000000         --exec-opt native.cgroupdriver=systemd'
+
+	Once you have editied the /etc/sysconfig/docker, execute following commands:
+
+		#systemctl daemon-reload
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Missing the space after #

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 33]
```diff
@@ -0,0 +1,474 @@
...
+
+		#systemctl stop docker
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Missing the space after #

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 34]
```diff
@@ -0,0 +1,474 @@
...
+
+		#systemctl start docker
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Missing the space after #

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 35]
```diff
@@ -0,0 +1,474 @@
...
+
+	For details, please refer
+    	https://docs.docker.com/engine/reference/commandline/dockerd/#examples
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Please link to the Oracle documentation:  https://docs.oracle.com/cd/E52668_01/E87205/html/section_yqk_dqg_gp.html

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 36]
```diff
@@ -0,0 +1,474 @@
...
+
+		net.core.wmem_max = 1048576
+
+		net.core.wmem_default = 262144
+
+		net.core.rmem_default = 262144
+
+	Execute following once the file is modified.
+
+		#sysctl -a
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Missing a space after #

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 37]
```diff
@@ -0,0 +1,474 @@
...
+
+		#sysctl -p
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Missing a space after #

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 38]
```diff
@@ -0,0 +1,474 @@
...
+	
+- Verify you have enough memory and cpu resources available for container. Each container for RAC requires 8GB memory and 16GB swap.  For details, please refer to the install guide for Grid Infrastructure and database for your version.
+
+- The Oracle RAC docker file, does not contain any Oracle Software Binaries.  Download the following software from the Oracle Technology Network (OTN) and stage them under (DOCKER_RAC_IMAGE)/dockerfile/(VERSION) folder, where (DOCKER_RAC_IMAGE) refers to the location where you unzipped the Oracle RAC docker file.
+	- Oracle Database 12c Release 2 Grid Infrastructure (12.2.0.1.0) for Linux x86-64
+	- Oracle Database 12c Release 2 (12.2.0.1.0) Enterprise Edition for Linux x86-64
+
+- You must have downloaded the patch# p27383741_122010_Linux-x86-64.zip from OTN as per the instruction given under "FOLLOWING FREELY AVAILABLE PATCH IS REQUIRED FOR THIS IMAGE TO WORK" heading. Stage it under (DOCKER_RAC_IMAGE)/dockerfile/(VERSION) folder.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: This is the right place to mention the patch. You can remove the stuff at the top of the README as it's not necessary to be that loud and vocal about it. It's just another file that's required.

File: OracleDatabase/OracleRealApplicationClusters/README.md [hunk 39]
```diff
@@ -0,0 +1,474 @@
...
+
+		- Patch# p27383741_122010_Linux-x86-64.zip
+			
+If you have not downloaded the patch, You can download this patch from following location:
+        
+  	 http://www.oracle.com/technetwork/database/database-technologies/clusterware/downloads/docker-4418413.html
+    
+  	Note: This patch required to run RAC on Docker in non-privileged mode. Do not unzip file file.
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/README.md]: Remove duplicate "file"

File: OracleDatabase/OracleRealApplicationClusters/dockerfile/12.2.0.1/Dockerfile [hunk 40]
```diff
@@ -0,0 +1,147 @@
...
+# Copy binaries
+# -------------
+# COPY Binaries
+COPY $INSTALL_FILE_1 $SETUPDOCKERORACLEINIT $INSTALL_GRID_PATCH $SETUP_LINUX_FILE $GRID_PATCH $GRID_SETUP_FILE $INSTALL_GRID_BINARIES_FILE $FIXUP_PREQ_FILE $INSTALL_FILE_2 $DB_SETUP_FILE $CHECK_SPACE_FILE $DB_INSTALL_RSP $INSTALL_DB_BINARIES_FILE $ENABLE_RAC_FILE $INSTALL_SCRIPTS/
+
+# Setup Scripts
+COPY $RUN_FILE $ADDNODE $ADDNODE_RSP $SETUPSSH $FUNCTIONS $CONFIGGRID $GRID_INSTALL_RSP $DBCA_RSP $PWD_FILE $CHECK_DB_FILE $USER_SCRIPTS_FILE $STOP_FILE $CHECK_DB_FILE $REMOTE_LISTENER_FILE $SETUPGRIDENV $SCRIPT_DIR/ 
+
+
+RUN chmod 777 $INSTALL_DIR/install/*.sh && \
```
Comment (Djelibeybi) [inline OracleDatabase/OracleRealApplicationClusters/dockerfile/12.2.0.1/Dockerfile]: Use `755`

File: OracleDatabase/OracleConnectionManager/README.md [hunk 41]
```diff
@@ -0,0 +1,56 @@
+Oracle Connection Manager on Docker
+
```
[the remark this answers is not in the document]
Comment (psaini79) [reply OracleDatabase/OracleConnectionManager/README.md]: Removed

**Pull request merged**